### DISCUSSION TITLE
Drag and drop should position the element such that the mouse is at the center.

### DISCUSSION THREAD

**Discussion by nhatduong-agilityio** [xyflow/xyflow#3038] [Q&A] [answered]
With a specific size. I want after dropping a node on the canvas the mouse is at the center of node.

**Comment by moklick** [comment, marked as answer]
You can use `nodeOrigin` for that. If you set it to `[0.5, 0.5]`, the position of a node is the center position https://reactflow.dev/docs/api/react-flow-props/#nodeorigin
